### ISSUE TITLE
error AL0572: I/O operations on the file or folder resulted in an exception

### ISSUE THREAD

**Issue by fvet** [microsoft/AL#5419]
Difficult to reproduce, but when packaging our project (5200+ files) I now and then get the error below:

`error AL0572: I/O operations on the file or folder './Layouts/NVT Orderbook.rdlc' resulted in an exception with the Windows 32 error code '0x000004C8'.`

It always refers to 'some' rdlc file (mostly a different one). These rdlc's, neither the parent report objects have been modified before packing / publishing. The files are also not opened. When repackaging / closing VS code, the error mostly does no longer occur.

![image](https://user-images.githubusercontent.com/25268332/68285396-80349e00-007f-11ea-972e-69e8d9ac0626.png)

**Comment by atoader** [comment]
Thanks @fvet ! We are aware of the problem and we are working on identifying the root cause.

**Comment by SirBETE** [comment]
Same problem here, using BC15CU1, AL langauge : 4.0.209944
error AL0572: I/O operations on the file or folder './REPORT/RDLC/R_20490_t.rdlc' resulted in an exception with the Windows 32 error code '0x000004C8'.
project consist of 855 files (not all AL files)

**Comment by fvet** [comment]
FYI, haven't seen the error a long time since using an updated insider AL extension...

![image](https://user-images.githubusercontent.com/25268332/72073469-40dd3480-32f0-11ea-910b-fd5ac1f59fbf.png)

**Comment by SirBETE** [comment]
> FYI, haven't seen the error a long time since using an updated insider AL extension...
> 
> ![image](https://user-images.githubusercontent.com/25268332/72073469-40dd3480-32f0-11ea-910b-fd5ac1f59fbf.png)


Could always try with that version ...

Just wondering how I can get that version, cause I doubt we are an bcinsider 
(time to disturb some colleagues! :D)

**Comment by SirBETE** [comment]
> > FYI, haven't seen the error a long time since using an updated insider AL extension...
> > ![image](https://user-images.githubusercontent.com/25268332/72073469-40dd3480-32f0-11ea-910b-fd5ac1f59fbf.png)
> 
> Could always try with that version ...
> 
> Just wondering how I can get that version, cause I doubt we are an bcinsider
> (time to disturb some colleagues! :D)

We have access, but using that compiler made it worse ...
The AL server crashed 5 times in the last 3 minutes. The server will not be restarted.

 Connection to server got closed. Server will restart.
Unhandled exception. System.TypeInitializationException: The type initializer for 'Microsoft.Dynamics.Nav.Deployment.Telemetry.TelemetryServiceManager' threw an exception.
'Microsoft.Dynamics.Nav.Deployment.Telemetry.TelemetryServiceManager' threw an exception.
 ---> System.TypeInitializationException: The type initializer for 'Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing.CoreEventSource' threw an exception.
 ---> System.ArgumentException: External component has thrown an exception.

So doubt it works ... :s

**Comment by SirBETE** [comment]
@atoader 

for : Error: An internal error has occurred
Log((Windows) has some info:

[2020-01-09 15:59:47.645] [renderer1] [error] Cannot read property 'kind' of null: TypeError: Cannot read property 'kind' of null
	at asHoverContent (c:\Users\me\.vscode\extensions\ms-dynamics-smb.al-4.0.218468\node_modules\vscode-languageclient\lib\protocolConverter.js:101:27)
	at asHover (c:\Users\me\.vscode\extensions\ms-dynamics-smb.al-4.0.218468\node_modules\vscode-languageclient\lib\protocolConverter.js:134:31)

**Comment by NKarolak** [comment]
I have just encountered the very same bug (as reported by fvet) in AL Language extension version 4.0.206840 while customizing the base application (BC 15.2 on-prem). 
Using the current 5.0 version from Marketplace seemed to fix it.

**Comment by itsme112358** [comment]
Got the same error today on 6.6.6.40897, targeting BC17 CU5. Recompiled three times without changing anything, then it worked 🤷‍♀️

**Comment by itsme112358** [comment]
Now I tried publishing --> extension gets compiled again --> same error, this is getting annoying. The repo is already excluded from Antivirus plus rdlc files are excluded in general, so it shouldn't interfere.
I have two extensions in a VSCode Workspace, maybe this causes the issue?
First try: <Don't remember which it was, but was an rdlc as well>
Second try: 
BOMCostShareDistribution.Report.al(4,18): error AL0572: I/O operations on the file or folder './BOMCostShareDistribution.rdlc' resulted in an exception with the Windows 32 error code '0x000004C8'. _(Base Application)_
Third try:
CloseInventoryPeriodTest.Report.al(4,18): error AL0572: I/O operations on the file or folder './CloseInventoryPeriodTest.rdlc' resulted in an exception with the Windows 32 error code '0x000004C8'. _(Base Application)_
Fourth try:
SCalculateNotAllowaInputTax.Report.al(8,18): error AL0572: I/O operations on the file or folder './RELion/RELion/S-Calculate NotAllowaInputTax.rdlc' resulted in an exception with the Windows 32 error code '0x000004C8'. _(Custom Object)_
Fifth try: Works!
I remember having had the issue a while back, then I only had to recompile once so I didn't care too much, but in this state the compiler is getting unusable.
Here is a list of my antivirus exclusions:
![image](https://user-images.githubusercontent.com/16080908/113112083-1931ff00-9209-11eb-886a-a6690238c608.png)

**Comment by spankau537826** [comment]
I still get the same error  when I run our build pipeline in devops when there's an Excel report layout with a pivot table in the project:

> error AL0572: I/O operations on the file or folder '.\reportext\CustomerItemSalesLayout.xlsx' resulted in an exception with the Windows 32 error code '0x00000003'.

When I remove the pivot table within the excel file, the pipeline runs fine. I also didn't get the same error when I published the app directly from VS code into the customer sandbox. If the problem is within the pipeline (which basically just creates a container on a temp VM to compile the app against), any idea where I could start looking?
Sorry if this is the wrong place to ask.

**Comment by JeremyVyska** [comment]
Just because this is one of the only issues that comes up when looking for this error:

I found I was getting this error for an On-Prem 20.0 version with various other report layouts during pipeline use.  Higher versions of the 20.x release resolve the issue.   

In my case, the on-prem client was due to upgrade to Cloud soon anyway (not bothering with a technical upgrade), so testing confirmed that I could simply build the pipeline to use the higher build and the app still functioned correctly on deployment (PTE, so not a runtime).
